### DISCUSSION TITLE
Device addition?

### DISCUSSION THREAD

**Discussion by msmith000** [libdyson-wg/ha-dyson#193] [Q&A]
Hey everyone,

I have the Dyson Purifier Cool Formaldehyde fan, but for the life of me I cannot seem to add the device itself to the integration.  I have used the account registration method as well as the "sticker" method, to no avail. 

The logs show the following error:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 819, in async_unload
    result = await component.async_unload_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/dyson_local/__init__.py", line 181, in async_unload_entry
    device: DysonDevice = hass.data[DOMAIN][DATA_DEVICES][entry.entry_id]
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
KeyError: '01J67HPXY8Q1CKGKV5BAPC8WEW'

Could I possibly get a suggestion to resolving this ?

**Comment by dotvezz** [comment]
Hey @msmith000! Is this still happening on your setup? It seems odd, like there's a mismatch between the Home Assistant config and the device itself.

**Reply by msmith000** [reply]
Yes, this is still the case.  I don’t understand what the issue could be.  Can you help me troubleshoot?
